File: OCaml Basics/5532.ml
open Scanf
open Printf
;;
let ceil a b =
  if a mod b = 0 then a/b else a/b+1
let f l a b c d =
  l - max (ceil a c) (ceil b d)
;;
scanf "%d\n%d\n%d\n%d\n%d\n"
  (fun l a b c d -> printf "%d\n" (f l a b c d))
;;

File: F# Basics/5532.fs
open System

let ceil a b =
    if a%b=0 then a/b else a/b+1

[<EntryPoint>]
let main _ =
    
    let l = Console.ReadLine() |> int
    let a = Console.ReadLine() |> int
    let b = Console.ReadLine() |> int
    let c = Console.ReadLine() |> int
    let d = Console.ReadLine() |> int

    printfn "%d" (l - max (ceil a c) (ceil b d))

    0
